tempesta_fw/http_parser.c

```diff
@@ -3825,7 +3239,7 @@ tfw_http_parse_req(void *req_data, unsigned char *data, size_t len,
 				   && TFW_LC(*(p + 6)) == 't'
 				   && *(p + 13) == ':'))
 			{
-				parser->_i_st = Req_HdrAcceptV;
+				parser->_i_st = &&Req_HdrAcceptV;
```

Comment:
The half of the `parse_req|resp` functions is parsing known headers char-by-char or bulk parsing like on these lines. Can we move that parsing to separate function like `parser_get_header()` which could collect the header.
PROs: 
- the whole function will be much shorter
- Less jumps in one function. Not really important since the patch implements direct jumps.
- The most important: the function can be generated ac compile time, from very simple description: header name and target label.
It's a general proposal, not for this patchset.

Reply:
Yes, definitely. In #309 we at least need to split the parser functions into headers and body parts. I saw that even with `barrier()` (the improvement is coming soon in the branch) GCC generates poor code layout, so smaller parsing functions are better.

The only confusing thing is how to call them? It seems we need additional branching for this, like
```
if (state is header parsing)
    parse_headers();
else if (state is body parsing)
    parse_body();
else
    some_error();
```

We already do the FSM split by our internal (`i`) states, e.g. to process certain header values.

How we can efficiently move the curacter-by-caracter logic out of the main logic?